quaint/src/visitor/query_writer.rs

```diff
@@ -0,0 +1,33 @@
+use crate::Value;
+use query_template::{Fragment, QueryTemplate};
+use std::fmt;
+use std::fmt::Display;
+
+pub(crate) trait QueryWriter {
+    fn write_string_chunk<D: fmt::Display>(&mut self, s: D);
+    fn write_parameter(&mut self);
+    fn write_parameter_tuple(&mut self);
+}
+
+impl QueryWriter for QueryTemplate<Value<'_>> {
+    fn write_string_chunk<D: Display>(&mut self, value: D) {
+        match self.fragments.last_mut() {
+            Some(Fragment::StringChunk(chunk)) => {
+                chunk.push_str(&value.to_string());
+            }
+            _ => {
+                let mut chunk = String::with_capacity(4096);
+                chunk.push_str(&value.to_string());
```

Comment:
An edge case but: this will lead to three allocations in total if `value.to_string()` is larger than 4 KB. This would be a more efficient way to do the same thing:

```suggestion
                let mut chunk = value.to_string();
                chunk.reserve(4096usize.checked_sub(chunk.len()).unwrap_or(0));
```

But, again, I'm not convinced that 4096 is a good number, or that we need this optimization at all.

Reply:
Simplified, pre-allocation removed. See above in the previous comment.